tests/integration/long/test_failure_types.py

```diff
@@ -301,3 +302,75 @@ def test_user_function_failure(self):
             """
             DROP TABLE test3rf.d;
             """, consistency_level=ConsistencyLevel.ALL, expected_exception=None)
+
+
+class TimeoutTimerTest(unittest.TestCase):
```

Comment:
Might be good to do some requests with callbacks and make sure the errback is called on timeout (this is new behavior with this change).

This could possibly be done as a standard integration test with a really short timeout (standard/test_cluster, maybe?) to get some coverage there.